modules/search/instant-search/components/search-result-minimal.jsx

```diff
@@ -31,14 +33,42 @@ const ShortcodeTypes = {
 };
 
 class SearchResultMinimal extends Component {
-	arrayOverlap( a1, a2 ) {
-		if ( ! Array.isArray( a1 ) ) {
-			a1 = [ a1 ];
-		}
-		const intersection = a1.filter( value => a2.includes( value ) );
-		return intersection.length !== 0;
+	componentDidMount() {
+		recordTrainTracksRender( this.getCommonTrainTracksProps() );
+	}
+
+	getCommonTrainTracksProps() {
+		return {
+			fetch_algo: 'jetpack-instant-search-api/v1',
+			fetch_position: this.props.index,
+			fetch_query: this.props.query,
+			railcar: this.props.railcarId,
+			rec_blog_id: this.props.result.fields.blog_id,
+			rec_post_id: this.props.result.fields.post_id,
+			ui_algo: 'jetpack-instant-search-ui/v1',
+			ui_position: this.props.index,
+		};
 	}
 
+	onClick = event => {
+		// User-triggered event
+		if ( event.isTrusted ) {
+			event.stopPropagation();
+			event.preventDefault();
+			// Send out analytics call
+			recordTrainTracksInteract( this.getCommonTrainTracksProps() );
+			// Await next animation frame to ensure w.js processes the queue
```

Comment:
I dug into `w.js`'s source and found that `STQ.prototype.push` is synchronous, so I figure we just need to give the browser enough time to initiate the network request for .t.gif`.

That said, I'd much prefer a function like `awaitNetworkRequestCompletion` over `requestAnimationFrame` for this use-case..

Reply:
> we just need to give the browser enough time to initiate the network request for t.gif.

considering the "try to send the pixel" is synchronous ( if `w.js` has been loaded ), why do we need to do that? Would the event bubbling / propagating by default have some kind of weird execution precedence over it? I think I don't understand why we're stoping the propagation, waiting for the synchronous part of `_tkq.push` to execute, waiting for the next tick, re-triggering the event, what does it buy us?

Reply:
Ah, I was under the incorrect impression that the default event action and the event handler would race. 

Since the default action typically executes after the event handler, I'll go ahead and simplify this function.